src/overview/container.js

```diff
@@ -40,14 +53,74 @@ class OverviewContainer extends Component {
         this.inputQueryEl = element
     }
 
+    renderNewTagOption() {
+        const { newTag } = this.props
+        if (newTag.length !== 0) {
+            return (
+                <TagOption
+                    data={newTag}
+                    active={false}
+                    newTag={1}
+                    addTagsToReverseDoc={this.props.addTags}
+                />
+            )
+        }
+        return null
+    }
+
+    renderTagsOptions() {
+        const { resultTags, newTag, deleteTags, suggestedTags } = this.props
+
+        if (resultTags.length === 0 && newTag.length === 0) {
```

Comment:
This type of stuff can go in a really simple memoized selector (use the `createSelector` function and basically put this condition in the body), to clean up the container code a bit and reduce what needs to be done every re-render.